core/src/test/java/org/opensearch/sql/data/model/DateTimeValueTest.java

```diff
@@ -95,6 +95,12 @@ public void dateValueInterfaceTest() {
     assertEquals("invalid to get dateValue from value of type INTEGER", exception.getMessage());
   }
 
+  @Test
+  public void dateValueInterfaceSupportedFormatTest() {
+    ExprValue dateValue = new ExprDateValue("03-Jan-21");
+    assertEquals(LocalDate.of(2021, 1, 3), dateValue.dateValue());
+  }
```

Comment:
Let's cover all the patterns and also the pattern not defined.

Reply:
Yes, added more tests covering the patterns and unsupported ones.